cpp/include/raft/neighbors/ivf_pq.cuh

```diff
@@ -230,11 +230,180 @@ auto build(raft::device_resources const& handle,
   return detail::build(handle, params, dataset, n_rows, dim);
 }
 
+/**
+ * @brief Unpack `n_take` consecutive records of a single list (cluster) in the compressed index
+ * starting at given `offset`, one code per byte (independently of pq_bits).
+ *
+ * Usage example:
+ * @code{.cpp}
+ *   // We will unpack the fourth cluster
+ *   uint32_t label = 3;
+ *   // Get the list size
+ *   uint32_t list_size = 0;
+ *   raft::copy(&list_size, index.list_sizes().data_handle() + label, 1, res.get_stream());
+ *   res.sync_stream();
+ *   // allocate the buffer for the output
+ *   auto codes = raft::make_device_matrix<float>(res, list_size, index.pq_dim());
+ *   // unpack the whole list
+ *   ivf_pq::unpack_list_data(res, index, codes.view(), label, 0);
+ * @endcode
+ *
+ * @tparam IdxT type of the indices in the source dataset
+ *
+ * @param[in] res
+ * @param[in] index
+ * @param[out] out_codes
+ *   the destination buffer [n_take, index.pq_dim()].
+ *   The length `n_take` defines how many records to unpack,
+ *   it must be smaller than the list size.
+ * @param[in] label
+ *   The id of the list (cluster) to decode.
+ * @param[in] offset
+ *   How many records in the list to skip.
+ */
+template <typename IdxT>
+void unpack_list_data(raft::device_resources const& res,
```

Comment:
Can we please start moving additional functions like this and the others in this PR into a different namespace like `raft::neighbors::ivf_pq::helpers`? I'm also suggesting that we do this with the `kmeans` API and other APIs that provide functions in addition to the standard operations. 

These new helper functions are extremely useful, but i also wonder if it's sustainable for us to keep adding these helper functions to the same public API headers- they run the risk of growing without bound and it makes them harder to navigate. What do you think about establishing a convention to put extra helpers like these in a separate header like `raft/neighbors/ivf_pq_helpers.cuh`? We could do the same w/ kmeans `raft/cluster/kmeans_helpers.cuh`. From what I've gathered, most of the helper functions should not need to call any of the non-helper public API functions so they *should* be pretty independent.